### ISSUE TITLE
Redis Version is wrongly parsed

### ISSUE THREAD

**Issue by thepirat000** [StackExchange/StackExchange.Redis#458]
When running a beta Redis Server (i.e. Redis on Windows) whose version string returned by the command INFO SERVER is not in the format **major.minor[.build[.revision]]**, the library assummes 2.0 Redis version.

I'm running Redis on windows, and the server is returning the following string as the Redis version:

**3.0.300-beta1**

And when SE.Redis tries to parse the version string (ResultProcessor.cs):

```
Version version;
if (Version.TryParse(val, out version))
{
    server.Version = version;
    server.Multiplexer.Trace("Auto-configured version: " + version);
}
```

Will fail to parse the version numbers, because of the **-beta1** part of the version string. 

After that, the library assumes a lower version of the server and consequently some commands like HSCAN as disabled.

Should the library remove the trailing **-xxxx** before parsing the version?

**Comment by NickCraver** [comment]
Sorry, we're crazy busy here on a few things and I'm packing up to move - so we're just been very delayed. This is totally legit, we'll take a look. Can you please drop a link to a beta build here we can test a fix against?

**Comment by thepirat000** [comment]
Thanks!
Here is the **3.0.300-beta1** version build:
https://github.com/MSOpenTech/redis/releases/tag/win-3.0.300-beta1

**Comment by NickCraver** [comment]
Sorry this took a while to get to, but looking at the state of the world:
1. OpenTech redis has closed down (sadly), but is on `3.2.100` as their latest release
2. Redis itself doesn't use alpha tags, just a version number

If MSOpenTech was still around and shipping there I'd agree we need a fix, but since they aren't and no one else needs it, I'm going to close this out. If it somehow does become a relevant issue in the future, we'll change up the parsing :)
